lib/event_sourcery/aggregate_root.rb

```diff
@@ -124,8 +124,12 @@ def load_history(events)
 
     def apply_event(event_class, options = {})
       event = event_class.new(**options.merge(aggregate_id: id))
-      mutate_state_from(event)
-      @changes << event
+      if event.valid?
```

Comment:
Not sure I'm following? The way I saw it is its an exceptional thing if it encounters an invalid event and therefore should 
- throw an error from the command side
- halt the process if emitting from a reactor (stop the line)

If you are sinking a collection of events, it would validate each event before committing any of them.